tests/settings.py

```diff
@@ -18,6 +40,7 @@
 
 
 middlewares = list(global_settings.MIDDLEWARE_CLASSES)
+middlewares.append('tests.settings.ShortCircuitMiddleware')
```

Comment:
you don't need a global middleware in settings - it would be better put it in the `test_middleware` file and `@modify_settings` used on just that test, to make it clear how that test's environment is different